rfcs/InputUnion.md

```diff
@@ -103,6 +103,30 @@ In addition, relying on this layer of abstraction means that this domain must be
 }
 ```
 
+Another approach is to use an input type with a discriminator and input fields for all possible member types.
+
+```graphql
+mutation {
+  logAnimalDropOff(
+    location: "Portland, OR"
+    animals: [
+      {type: CAT, name: "Buster", age: 3, livesLeft: 7},
+      {type: DOG, name: "Ripple", age: 2, breed: WHIPPET}
+    ]
+  )
+}
+
+input AnimalDropOffInput {
+  type: AnimalType!
+  name: String!
+  age: Int!
+  breed: DogBreed # only applies to dogs
```

Comment:
How is the mapping between the field `breed` and `type: DOG` communicated?

Seems like this particular example is already possible, right?

Reply:
The information that the `breed` input field only applies when `type = DOG` and the `livesLeft` input field only applies when `type = CAT` is only conveyed via human readable documentation which is the major shortcoming with this approach. This schema information isn't conveyed in a machine, readable format that can be used by tooling to ensure clients only submit valid combinations of input fields for a given `type. It's really the same drawbacks as using "tagged unions" but with a slightly different literal shape.

On a somewhat related note, I'll update this example input type to include snake input fields to be consistent with the earlier example.

Reply:
My second question is: how is this different than what you can already do with GraphQL as-is?

Tagged unions _do_ introduce something you can't do today & has a direct machine understandable mapping from type to fields...

Reply:
You can absolutely do this in GraphQL as-is! The proposed content is in the "Problem Sketch" section of the RFC which describes how people solve input polymorphism problems with GraphQL today and the shortcomings of these workarounds. The "tagged union" I was referring to is the current tagged union [workaround](https://github.com/graphql/graphql-spec/blame/master/rfcs/InputUnion.md#L61) and not the proposed [first class tagged union](https://github.com/graphql/graphql-spec/blame/master/rfcs/InputUnion.md#L778). Does that clear up any confusion?

Reply:
oops, yeah it does clear it up - sorry the doc has gotten big and i assumed this was in the solutions section...